### DISCUSSION TITLE
Heartbeat config: timeout or interval?

### DISCUSSION THREAD

**Discussion by shadz3rg** [php-amqplib/php-amqplib#1102] [General]
Hello community,
as [RabbitMQ docs](https://www.rabbitmq.com/heartbeats.html#heartbeats-interval) states:

> Heartbeat frames are sent about every heartbeat timeout / 2 seconds. This value is sometimes referred to as the heartbeat interval. After two missed heartbeats, the peer is considered to be unreachable. Different clients manifest this differently but the TCP connection will be closed. When a client detects that RabbitMQ node is unreachable due to a heartbeat, it needs to re-connect. 

Division by 2 hinting on timeout:
https://github.com/php-amqplib/php-amqplib/blob/99374c1d4dd8a362bf7b0dcd4fd96b231170f439/PhpAmqpLib/Wire/IO/AbstractIO.php#L151

But when value is multiplied by two (like it is an interval? 🤔):
https://github.com/php-amqplib/php-amqplib/blob/99374c1d4dd8a362bf7b0dcd4fd96b231170f439/PhpAmqpLib/Wire/IO/AbstractIO.php#L165

**Comment by ramunasd** [comment]
You see different cases - 1st checks if we should send heartbeat to broker, while 2nd check when last activity from broker was. As You quoted _"After two missed heartbeats, the peer is considered to be unreachable"_, that's why multiple by 2.
